src/sql/workbench/contrib/query/test/browser/queryInputFactory.test.ts

```diff
@@ -141,6 +141,217 @@ suite('Query Input Factory', () => {
+		instantiationService.stub(IObjectExplorerService, new MockObjectExplorerService());
+		instantiationService.stub(IConnectionManagementService, connectionManagementService);
+		instantiationService.stub(IEditorService, editorService);
+		const queryEditorLanguageAssociation = instantiationService.createInstance(QueryEditorLanguageAssociation);
+		const input1 = createFileInput(URI.file('/test/file.sql'), undefined, undefined, undefined);
+		const input2 = createFileInput(URI.file('/test/file.sql'), undefined, undefined, undefined);
+		queryEditorLanguageAssociation.syncConvertInput(input1);
+		queryEditorLanguageAssociation.syncConvertInput(input2);
+		let connProfile = connectionManagementService.getConnectionProfile('file:///test/file.sql');
+		assert(connProfile !== undefined, 'connection profile should not be undefined');
+		assert(connProfile.azureResourceId === 'file:///test/file.sql', 'azureResourceId should not be undefined');
+		assert(connectionManagementService.numberConnects === 1, 'Convert input should have called connect when active OE connection exists');
+	});
+
+	test('sync query editor inputs can be connected one after the other when global connection exists', () => {
+		const instantiationService = workbenchInstantiationService();
+		const editorService = new MockEditorService();
+		const connectionManagementService = new MockConnectionManagementService();
+		instantiationService.stub(IObjectExplorerService, new MockObjectExplorerService());
+		instantiationService.stub(IConnectionManagementService, connectionManagementService);
+		instantiationService.stub(IEditorService, editorService);
+		const queryEditorLanguageAssociation = instantiationService.createInstance(QueryEditorLanguageAssociation);
+		const input1 = createFileInput(URI.file('/test/file1.sql'), undefined, undefined, undefined);
+		const input2 = createFileInput(URI.file('/test/file2.sql'), undefined, undefined, undefined);
+		queryEditorLanguageAssociation.syncConvertInput(input1);
+		queryEditorLanguageAssociation.syncConvertInput(input2);
+		let connProfile1 = connectionManagementService.getConnectionProfile('file:///test/file1.sql');
+		assert(connProfile1 !== undefined, 'connection profile should not be undefined');
+		assert(connProfile1.azureResourceId === 'file:///test/file1.sql', 'azureResourceId should not be undefined');
+		let connProfile2 = connectionManagementService.getConnectionProfile('file:///test/file2.sql');
+		assert(connProfile2 !== undefined, 'connection profile should not be undefined');
+		assert(connProfile2.azureResourceId === 'file:///test/file2.sql', 'azureResourceId should not be undefined');
+		assert(connectionManagementService.numberConnects === 2, 'Convert input should have called connect two times when active OE connection exists and we have two inputs');
+	});
+
+	test('query editor input can only be connected once when global connection exists (OE)', async () => {
```

Comment:
What is the OE part referring to? As far as I can tell this test is doing the same thing as the others (it's just using convertInput instead of syncConvertInput)

Same with the "Editor" tests below

Reply:
Just following what the other tests were doing above, which I don't know why that's the case.

Reply:
Only difference between the Editor and Object Explorer tests are that the Object Explorer tests use a mock Object Explorer Service Test, while the Editor ones do not.